### ISSUE TITLE
GetGDCMSeriesFileNames could not find Series after set useSeriesDetails to True.

### ISSUE THREAD

**Issue by blacksino** [SimpleITK/SimpleITK#1809]
Hello,when I set useSeriesDetails to False, series could be found with the following code:
`series_file_names = series_reader.GetGDCMSeriesFileNames(self._dcm_dir, series_IDs[i])`
however, it returns 
`WARNING: In /tmp/SimpleITK-build/ITK/Modules/IO/GDCM/src/itkGDCMSeriesFileNames.cxx, line 136
GDCMSeriesFileNames (0x557f56e295e0): No Series were found`
when I use
`series_reader.GetGDCMSeriesFileNames(self,_dcm_dir, series_IDs[i],useSeriesDetails=True)`.

How could this be?BTW, the intention of using useSeriesDetails is that there are multi-series with the same seriesID.

**Comment by zivy** [comment]
Hello @blacksino,

Thanks for the bug report. Was able to reproduce. This looks like a bug in the underlying ITK or GDCM layers. Will investigate further and update.

**Comment by zivy** [comment]
After a bit of digging it appears this is a "feature" in ITK and not a bug. In SimpleITK this is a missing feature which is addressed by [this PR](#1819). The `useSeriesDetails` flag has to have the same value for the calls to `GetGDCMSeriesIDs` and `GetGDCMSeriesFileNames`.

Code illustrating usage:
```
import SimpleITK as sitk

dcm_dir = "data/"

use_series_details = True
series_IDs = series_reader.GetGDCMSeriesIDs(dcm_dir, use_series_details)
series_file_names = series_reader.GetGDCMSeriesFileNames(dcm_dir, series_IDs[0], use_series_details)
```

**Comment by zivy** [comment]
PR merged, resolving the issue.
